### PR TITLE
fix(emails): fall back to ui-avatars.com when no avatar

### PR DESCRIPTION
**Pull request by xdamman**
Fixes https://github.com/opencollective/opencollective/issues/1906

### CODE REVIEW

File: server/models/Collective.js [hunk 1]
```diff
@@ -898,11 +899,12 @@ export default function(Sequelize, DataTypes) {
    * @param {*} startDate beginning of the time period
    * @param {*} endDate end of the time period
    */
-  Collective.prototype.getNewOrders = async function(startDate = 0, endDate = new Date()) {
```
Comment (znarf) [inline server/models/Collective.js]: Looks it's not related to the PR.

File: server/models/Collective.js [hunk 2]
```diff
@@ -221,6 +221,7 @@ export default function(Sequelize, DataTypes) {
             const image = `https://logo.clearbit.com/${getDomain(this.website)}`;
             return image;
           }
+          return `https://ui-avatars.com/api/?rounded=true&name=${encodeURIComponent(this.name)}`;
```
Comment (znarf) [inline server/models/Collective.js]: Shouldn't we only use ui-avatars for 'USER'?

Comment (znarf) [reply server/models/Collective.js]: What's the impact of that change in the frontend?

File: server/models/Collective.js [hunk 3]
```diff
@@ -898,11 +899,12 @@ export default function(Sequelize, DataTypes) {
    * @param {*} startDate beginning of the time period
    * @param {*} endDate end of the time period
    */
-  Collective.prototype.getNewOrders = async function(startDate = 0, endDate = new Date()) {
+  Collective.prototype.getNewOrders = async function(startDate = 0, endDate = new Date(), options = {}) {
     const orders = await models.Order.findAll({
       where: {
         CollectiveId: this.id,
         createdAt: { [Op.gte]: startDate, [Op.lt]: endDate },
+        ...options.where,
```
Comment (znarf) [inline server/models/Collective.js]: Looks not related to the PR.

File: server/models/Collective.js (hunk 2 above)
Comment (xdamman) [reply server/models/Collective.js]: Good idea to only do it for `USER`.

It means that we will have to check for `ui-avatars.com` instead of `undefined` if we want to show a different default avatar on the frontend. But in the meantime, it will fix this: https://github.com/opencollective/opencollective/issues/1903

File: server/models/Collective.js (hunk 1 above)
Comment (xdamman) [reply server/models/Collective.js]: ah right

File: server/models/Collective.js (hunk 3 above)
Comment (xdamman) [reply server/models/Collective.js]: removing this for cleanness.

**Comment by znarf** [comment]
I think we should fix that differently.

- The API should not be aware of `ui-avatars`, it's the responsibility of the `images` service.
- In the sake of simplification, we should also drop the SVG initials in frontend and only rely on `ui-avatars` (through the `images` service).

The responsibility of the API should then only be to expose the URL of the image service: `https://images.opencollective.com/{slug}/avatar.png`

**Comment by znarf** [comment]
I introduced some helpers in https://github.com/opencollective/opencollective-api/pull/1909

`collective.getImageUrl();`

**Comment by znarf** [comment]
Closing the PR. We will fix the issue differently.
